erigon-lib/commitment/hex_patricia_hashed.go

```diff
@@ -1879,6 +1879,115 @@ func (hph *HexPatriciaHashed) RootHash() ([]byte, error) {
 	return rootHash[1:], nil // first byte is 128+hash_len=160
 }
 
+func (hph *HexPatriciaHashed) GenerateTouchedKeyTrie(ctx context.Context, updates *Updates, expectedRootHash []byte, logPrefix string) (touchedKeyTree *trie.Trie, rootHash []byte, err error) {
```

Comment:
Not necessary to duplicate code as you can directly call `GenerateWitness` and get the desired result.